openai/openai.go

```diff
@@ -70,65 +102,44 @@ func (oac *OpenAiClient) GetCompletion(prompt string, debug bool) (*string, erro
 		fmt.Print(logMsg)
 	}
 
-	resp, err := oac.client.CreateChatCompletion(
-		context.Background(),
-		openai.ChatCompletionRequest{
-			Model: openai.GPT3Dot5Turbo,
-			Messages: []openai.ChatCompletionMessage{
-				{
-					Role:    openai.ChatMessageRoleSystem,
-					Content: systemMessage,
-				},
-				{
-					Role:    openai.ChatMessageRoleUser,
-					Content: prompt,
-				},
-			},
+	firstDraft, err := oac.RequestCompletion([]openai.ChatCompletionMessage{
+		{
+			Role:    openai.ChatMessageRoleSystem,
+			Content: systemMessage,
 		},
-	)
-
+		{
+			Role:    openai.ChatMessageRoleUser,
+			Content: prompt,
+		},
+	})
 	if err != nil {
 		return nil, err
 	}
 
-	fmt.Printf(
-		"PromptTokens:\t\t%d\nCompletionTokens:\t%d\nTotalTokens:\t\t%d\n",
-		resp.Usage.PromptTokens,
-		resp.Usage.CompletionTokens,
-		resp.Usage.TotalTokens,
-	)
-	firstDraft := &resp.Choices[0].Message.Content
-
-	resp2, err2 := oac.client.CreateChatCompletion(
-		context.Background(),
-		openai.ChatCompletionRequest{
-			Model: openai.GPT3Dot5Turbo,
-			Messages: []openai.ChatCompletionMessage{
```

Comment:
The GetCompletion method is a bit long and seems to be too nested. Consider refactoring the method for better readability and to make it easy to understand what is happening at each point.

Reply:
Thank you for your feedback! The AI gives your comment a score of 4